src/clientiface.cpp

```diff
@@ -192,17 +192,35 @@ void RemoteClient::GetNextBlocks (
 	*/
 	s32 new_nearest_unsent_d = -1;
 
-	// get view range and camera fov from the client
+	// Get view range and camera fov (radians) from the client
 	s16 wanted_range = sao->getWantedRange() + 1;
 	float camera_fov = sao->getFov();
 
-	const s16 full_d_max = std::min(adjustDist(m_max_send_distance, camera_fov), wanted_range);
-	const s16 d_opt = std::min(adjustDist(m_block_optimize_distance, camera_fov), wanted_range);
+	// If below the heuristic zoom threshold (see adjustDist() in numeric.cpp)
+	// distrust client-sent FOV and get server-set player object property
+	// zoom FOV (degrees) as a check to avoid hacked clients using FOV to load
+	// distant world.
+	// 0.888 radians is slightly larger than the zoom threshold of 1.775 / 2
+	// radians.
+	if (camera_fov < 0.888f) {
+		float prop_zoom_fov = sao->getZoomFOV();
+		// If zoom is disabled by value 0
+		if (prop_zoom_fov == 0.0f)
```

Comment:
Same as in `camera.cpp` to compensate calculation/conversion errors: `prop_zoom_fov < 0.001f`

Reply:
Ok.

Reply:
Done.